util/types/time.go

```diff
@@ -1184,13 +1210,22 @@ func parseDateTimeFromNum(num int64) (Time, error) {
 // The valid datetime range is from '1000-01-01 00:00:00.000000' to '9999-12-31 23:59:59.999999'.
 // The valid timestamp range is from '1970-01-01 00:00:01.000000' to '2038-01-19 03:14:07.999999'.
 // The valid date range is from '1000-01-01' to '9999-12-31'
-func ParseTime(str string, tp byte, fsp int) (Time, error) {
+func ParseTime(str string, tp byte, fst int) (Time, error) {
+	return parseTime1(str, tp, fst, false)
+}
+
+// ParseTimeFromFloatString is similar to ParseTime, except that it's used to parse a float converted string.
+func ParseTimeFromFloatString(str string, tp byte, fst int) (Time, error) {
+	return parseTime1(str, tp, fst, true)
+}
+
+func parseTime1(str string, tp byte, fsp int, isFloat bool) (Time, error) {
```

Comment:
why is there a `1`?

Reply:
rest LGTM

Reply:
Both ParseTime and ParseTimeFromFloatString call this function, would you suggest a better name?

Reply:
the inner function parseTime1(or parseDatetime) is not fully compatible with MySQL's str_to_datetime. I'm migrating this code.

Reply:
Yes, it may ont compatible with str_to_datetime, but the change doesn't conflict with what's going on.
Once your str_to_datetime is ready, we can use it to implement `ParseTime`.
What we should care is the **exported** functions, the interface rather than implementation, using tests to ensure correctness.

Reply:
shall we just call it `parseTime`? the lower case already indicated that it's a private func